sound/soc/intel/boards/bytcht_es8316.c

```diff
@@ -544,7 +545,16 @@ static int snd_byt_cht_es8316_mc_probe(struct platform_device *pdev)
 		props[cnt++] = PROPERTY_ENTRY_BOOL("everest,jack-detect-inverted");
 
 	if (cnt) {
-		ret = device_add_properties(codec_dev, props);
+		fwnode = fwnode_create_software_node(props, NULL);
+		if (IS_ERR(fwnode)) {
+			put_device(codec_dev);
+			return PTR_ERR(fwnode);
+		}
+
+		ret = device_add_software_node(codec_dev, to_software_node(fwnode));
+
+		fwnode_handle_put(fwnode);
```

Comment:
I;m not sure I understood why is that?

Reply:
no idea, that's how the code was initially written by Heikki. https://lore.kernel.org/alsa-devel/20210322110638.2681-1-heikki.krogerus@linux.intel.com/

````diff
+	fwnode = fwnode_create_software_node(props, NULL);
+	if (IS_ERR(fwnode)) {
+		put_device(sdw_dev);
+		return PTR_ERR(fwnode);
 	}
 
-	ret = device_add_properties(sdw_dev, props);
+	ret = device_add_software_node(sdw_dev, to_software_node(fwnode));
+
+	fwnode_handle_put(fwnode);
````

I assumed the fwnode_handle_put() was there for a reason....

Reply:
`fwnode_create_software_node()` returns a fwnode with a ref-count of 1, `device_add_software_node ()` takes a ref itself, the ref taken by `device_add_software_node()` will be released by `device_remove_software_node()`. The purpose of the `fwnode_handle_put(fwnode);` call here is to release the reference returned by `fwnode_create_software_node()`.